pkgs/by-name/s7/s7/package.nix

```diff
@@ -0,0 +1,141 @@
+{
+  lib,
+  fetchFromGitLab,
+  stdenv,
+  notcurses,
+  gmp,
+  mpfr,
+  libmpc,
+  flint3,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "s7";
+  version = "11.2-unstable-2024-10-29";
+
+  src = fetchFromGitLab {
+    domain = "cm-gitlab.stanford.edu";
+    owner = "bil";
+    repo = "s7";
+    rev = "15ff6156a0c836bda702cbf816c370daf9c83910";
+    hash = "sha256-anH9gPhAbxvv3XKb9/zEk4U8pRoXwVApWGu0S97oqbk=";
+  };
+
+  buildInputs = [
+    notcurses
+    gmp
+    mpfr
+    libmpc
+    flint3
+  ];
+
+  # The following scripts are modified from [Guix's](https://packages.guix.gnu.org/packages/s7/).
+
+  patchPhase = ''
+    runHook prePatch
+
+    substituteInPlace s7.c \
+        --replace-fail libc_s7.so $out/lib/libc_s7.so
+
+    runHook postPatch
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    cc s7.c -o s7-repl \
+        -O2 -I. \
+        -Wl,-export-dynamic \
+        -lm -ldl \
+        -DWITH_MAIN \
+        -DS7_LOAD_PATH=\"$out/share/s7/scm\"
+
+    cc s7.c -o s7-nrepl \
+        -O2 -I. \
+        -Wl,-export-dynamic \
+        -lm -ldl \
+        -DWITH_MAIN \
+        -DWITH_NOTCURSES -lnotcurses-core \
+        -DS7_LOAD_PATH=\"$out/share/s7/scm\"
```

Comment:
Is the first attempt at compilation useful for something?

Reply:
s7 has two REPLs. `WITH_MAIN` only it builds the older one, `WITH_NOTCURSES` in addition it builds the newer one. The older one has runtime dependencies, which is built with itself in the later commands.

Reply:
This is a good comment for in the derivation itself to be fair